pkg/kv/bulk/sst_batcher.go

```diff
@@ -585,18 +598,31 @@ func (b *SSTBatcher) doFlush(ctx context.Context, reason int) error {
 		}
 	}
 
+	currentBatchStatsCopy := b.currentBatchStats.Identity().(*backuppb.IngestionPerformanceStats)
+	currentBatchStatsCopy.Combine(&b.currentBatchStats)
+	b.currentBatchStats.Reset()
+
 	fn := func(ctx context.Context) error {
 		defer res.Release()
 		defer b.mem.Shrink(ctx, reserved)
-		if err := b.addSSTable(ctx, batchTS, start, end, data, stats, !flushAsync); err != nil {
+		if err := b.addSSTable(ctx, batchTS, start, end, data, stats, !flushAsync, currentBatchStatsCopy); err != nil {
 			return err
 		}
 		b.mu.Lock()
 		summary.DataSize += int64(size)
+		currentBatchStatsCopy.DataSize += int64(size)
```

Comment:
Is it safe to drop the atomic add because we're holding on to the b.mu lock while updating these stats?

Reply:
This should be okay to modify because of the same copy reason as above.

Reply:
Same comment as below, I agree we're operating on a copy but isn't that copy shared by concurrent invocations of `fn`?

Reply:
Ohh I think i understand. Even if we have concurrent invocation of `fn` across flushes, each one will have its own copy. If this is true, maybe a quick comment to this effect?